mysql-test/main/mdl_sync.test

```diff
@@ -3249,6 +3249,7 @@ connection con2;
 SET DEBUG_SYNC= 'now WAIT_FOR table_opened';
 --echo # Check that FLUSH must wait to get the GRL
 --echo # and let DROP PROCEDURE continue
+--source ../suite/innodb/include/wait_all_purged.inc
```

Comment:
This can make the test extremely slow if you do not first `SET GLOBAL innodb_purge_rseg_truncate_frequency=1;` like the include file says. Remember to save and restore the setting when you override it.

I do not think we need this in 10.4. Before ea37b144094a0c2ebfc6774047fd473c1b2a8658, the purge of committed InnoDB transaction history would only acquire MDL when the table contains indexed virtual columns. As far as I can tell, at this point no InnoDB tables have been created in this test. The only InnoDB table in this test would be `t2`, created some 25 lines later.